### DISCUSSION TITLE
Issue upgrading to dunfell-l4t-r32.5.0

### DISCUSSION THREAD

**Discussion by mbilloo** [OE4T/meta-tegra#663] [Q&A] [answered]
Hi,

We updated our meta-tegra layer from commit 387d97e to branch dunfell-l4t-r32.5.0. When we went to build our Yocto image, bitbake ran into the following error when parsing recipes:

[mbilloo@mbilloo-blues build]$ bitbake custom-image-base                                                                                                                 
Parsing recipes: 100% |#####################################################################################################################################################################################################################################################################################################################| Time: 0:00:18
Parsing of 2794 .bb files complete (0 cached, 2794 parsed). 4219 targets, 141 skipped, 0 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies                                   
**ERROR: Nothing PROVIDES 'cuda-binaries-ubuntu1804-native'**                    
cuda-binaries-ubuntu1804-native was skipped: Recipe requires NVIDIA_DEVNET_MIRROR setup
NOTE: Runtime target 'gstreamer1.0-plugins-tegra' is unbuildable, removing... 
Missing or unbuildable dependency chain was: ['gstreamer1.0-plugins-tegra', 'gstreamer1.0-plugins-nvvideosinks', 'cuda-cudart-native', 'cuda-binaries-ubuntu1804-native']
NOTE: Runtime target 'packagegroup-custom' is unbuildable, removing...      
Missing or unbuildable dependency chain was: ['packagegroup-custom, 'gstreamer1.0-plugins-tegra', 'gstreamer1.0-plugins-nvvideosinks', 'cuda-cudart-native', 'cuda-binaries-ubuntu1804-native']
ERROR: Required build target 'custom-image-base' has no buildable providers.                                                                                                  
Missing or unbuildable dependency chain was: ['custom-image-base', 'packagegroup-custom', 'gstreamer1.0-plugins-tegra', 'gstreamer1.0-plugins-nvvideosinks', 'cuda-cudart-native', 'cuda-binaries-ubuntu1804-native']

Can anybody point to what we're doing wrong?

Thanks

**Comment by dwalkes** [comment]
> cuda-binaries-ubuntu1804-native was skipped: Recipe requires NVIDIA_DEVNET_MIRROR setup

I think this is the hint that you need the instructions in https://github.com/OE4T/meta-tegra/wiki/L4T-R32.3.1-Notes#sdk-manager-downloads-required to setup the devnet mirror for cuda host tool binaries.

I've setup a new page at https://github.com/OE4T/meta-tegra/wiki/NVIDIA_DEVNET_MIRROR-and-SDK-Manager to document the fact that we still need this for CUDA host tool content, until we can take this off the L4T Integration issues list at https://github.com/OE4T/meta-tegra/wiki/L4T-Integration-Issues with help from nvidia.

**Comment by madisongh** [comment, marked as answer]
The earlier commit you mention was for L4T R32.4.3, and now you've upgraded to R32.5.0.  One of the changes in the update is that NVIDIA has released sources for several of the Gstreamer plugins that used to be binary-only, and the layer has been updated to build them from source. Some of those plugins use CUDA, though, which is why they now have a dependency on the CUDA host-side tools, where they didn't before.

**Reply by bkirkTag** [reply]
I don't know if this problem relates to what i have expierensed several times - my builds works for some weeks and then suddenly fails due to missing cuda-stuff (like libcudablas_xxxx). I looks like the NvidiaSDKmanager did some auto update in the background.
To make my build work again, I used the SDKmanager to force the target operating system back to Jetpack 4.5 (instead of 4.5.1) !

**Reply by dwalkes** [reply]
NVIDIA has recently provided an update which should mean this isn't required going forward.  See https://github.com/OE4T/meta-tegra/pull/677/files and updates to https://github.com/OE4T/meta-tegra/wiki/L4T-Integration-Issues and https://github.com/OE4T/meta-tegra/wiki/NVIDIA_DEVNET_MIRROR-and-SDK-Manager pages

**Comment by mbilloo** [comment]
Thanks! I ended up just removing the gstreamer plugin that was causing this trouble as a dependency in our final build image.
